### PR TITLE
Fix inconsistent of remote API and document

### LINKED ISSUE #12748: Inconsistent of remote API and document
remote API response as:

     {"description":"","is_official":false,"is_trusted":true,"name":"ariarijp/sshd","star_count":0}

But in document, it should be *is_automated* instead of *is_trusted*

### PR DESCRIPTION
**Pull request by HuKeping**
Fix #12748

Signed-off-by: Hu Keping hukeping@huawei.com

### CODE REVIEW

**Comment by linlinlinlin** [comment]
This commit reserve *is_trusted* and *is_automated* together ?

Does it have keep only one? or these two are represent different things.

**Comment by HuKeping** [comment]
@linlinlinlin

This is the output from the REST API on docker/master
```
[
    {
        "star_count": 13,
        "is_official": true,
        "name": "library/iam-proxy",
        "is_trusted": true,
        "is_automated": false,
        "description": ""
    }
]
```

**Comment by SvenDowideit** [comment]
LGTM - @jlhawn ?

**Comment by jlhawn** [comment]
@SvenDowideit The search API is not in my domain of knowledge. @kencochrane would probably know.

**Comment by calavera** [comment]
This change looks consistent with the API :+1: 

pinging @moxiegirl, @thaJeztah, @jamtur01 from the docs team.

**Comment by thaJeztah** [comment]
Did this output change recently? If not, it would be great to also fix older API docs (1.18, 1.17). 

Apart from that, LGTM (if others don't think it's nescessary, feel free to merge)

I have to agree with earlier
comments that it's odd to still have `is_trusted` in the output; I don't think that term is still used (because they're not *trusted*, just *automated*). If someone could create a separate issue for that?

**Comment by HuKeping** [comment]
I'll check that

**Comment by HuKeping** [comment]
hi @thaJeztah as far as I can tell  the output  under API v1.18 is like
```
{"description":"","is_official":false,"is_trusted":false,"name":"clintonskitson/distribution","star_count":0}
```
so yes, it was a recently change.

**Comment by thaJeztah** [comment]
Thanks!

Just did a search for the `is_trusted` Boolean, and according to this commit, it shouldn't be there anymore: https://github.com/docker/docker/commit/9fc8028c987ac560ceb3b29eb609d77a82abdacc

And should have been **replaced** with `is_automated`, so *somewhere* there's something that isn't right

@jlhawn @kencochrane if this is the output that's returned by Docker Hub; is that correct?

**Comment by kencochrane** [comment]
@thaJeztah if you search the API, and take the first result, this is what we are returning on v1 search API

https://index.docker.io/v1/search?q=ubuntu

```
{
  "is_automated": false,
   "name": "ubuntu",
   "is_trusted": false,
   "is_official": true,
   "star_count": 1678, 
   "description": "Official Ubuntu base image"
}
```

we return both ``is_trusted`` and ``is_automated``, they are the same value, we keep ``is_trusted`` around for backwards compatibility, but it will go away in new versions, so moving forward we should use ``is_automated``.

**Comment by thaJeztah** [comment]
@kencochrane thanks for clarifying, that makes sense.

I think that, since the API is versioned, it would be good to remove/strip the `is_trusted` property from the result in the Docker API, but that would be a separate issue.

@HuKeping could you add an extra `note:` that mentions that the `is_trusted` property is deprecated, and is only present for backward-compatibility?

There's already a note, so add an extra "note" like the one that's already there.

**Comment by thaJeztah** [comment]
^^ oh, and make a mention that people should use `is_automated` instead.

**Comment by calavera** [comment]
What's the status of this @HuKeping? Can you make the changes that @thaJeztah mentions? I can finish this PR is you're not interested in finishing it anymore.

**Comment by HuKeping** [comment]
@calavera  Oh so sorry about that...I've almost forget I have a PR here, will be done today.

**Comment by HuKeping** [comment]
@calavera @thaJeztah Please take a look if this change was ok

File: docs/sources/reference/api/docker_remote_api_v1.19.md [hunk 1]
```diff
@@ -1525,6 +1525,11 @@ Search for an image on [Docker Hub](https://hub.docker.com).
 > The response keys have changed from API v1.6 to reflect the JSON
 > sent by the registry server to the docker daemon's request.
 
+> **Note**:
```
Comment (moxiegirl) [inline docs/sources/reference/api/docker_remote_api_v1.19.md]: There are some English errors. Also , we don't want to stack notes like this.  We should make the  second note part of the description.   No need to mention we are keeping around `is_trusted` for backwards compatibility --- that is implied by the deprecation statement.

--> https://gist.github.com/moxiegirl/be45866159b906e3c773

Search for an image on [Docker Hub](https://hub.docker.com). This API
returns both `is_trusted` and `is_automated` images. Currently, they 
are considered identical. In the future, the `is_trusted` property will
be deprecated and replaced by the `is_automated` property.

> **Note**:
> The response keys have changed from API v1.6 to reflect the JSON
> sent by the registry server to the Docker daemon's request.

**Comment by HuKeping** [comment]
@moxiegirl  updated

**Comment by calavera** [comment]
Thanks @HuKeping! Waiting for a couple LGTM from @docker/docs-owners before merging.

**Comment by moxiegirl** [comment]
@HuKeping please move the second paragraph of the note into the description.  The information you provided is simply how this API works, not exceptional information.

Like this:

-->

Search for an image on [Docker Hub](https://hub.docker.com). This API
returns both `is_trusted` and `is_automated` images. Currently, they 
are considered identical. In the future, the `is_trusted` property will
be deprecated and replaced by the `is_automated` property.

> **Note**:
> The response keys have changed from API v1.6 to reflect the JSON
> sent by the registry server to the Docker daemon's request.

**Comment by HuKeping** [comment]
Thanks @moxiegirl , updated

File: docs/sources/reference/api/docker_remote_api_v1.19.md [hunk 2]
```diff
@@ -1517,7 +1517,10 @@ Status Codes:
 
 `GET /images/search`
 
-Search for an image on [Docker Hub](https://hub.docker.com).
+Search for an image on [Docker Hub](https://hub.docker.com).This API
```
Comment (thaJeztah) [inline docs/sources/reference/api/docker_remote_api_v1.19.md]: Missing a space before `This API`

**Comment by thaJeztah** [comment]
One really minor correction, which would be nice to fix, but no show-stopper.

LGTM

**Comment by HuKeping** [comment]
Thanks @thaJeztah , updated

**Comment by thaJeztah** [comment]
@HuKeping great!

ping @moxiegirl ptal

**Comment by moxiegirl** [comment]
Thanks for the contribution!

**Pull request merged**